document/src/vespa/document/select/parser_limits.cpp

```diff
@@ -0,0 +1,11 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+#include "parser_limits.h"
+#include "parsing_failed_exception.h"
+
+namespace document::select {
+
+void throw_max_depth_exceeded_exception() {
+    throw ParsingFailedException("expression is too deeply nested");
```

Comment:
Consider adding the depth and max depth limit to the exception message.

Reply:
Added limit to message. We cannot know the actual depth of the input AST as we terminate parsing immediately when hitting the limit.